### PR TITLE
RavenDB-17093  OLAP ETL: Allow to edit script name in studio

### PR DESCRIPTION
**Pull request by Danielle9897**

### CODE REVIEW

File: src/Raven.Studio/typescript/models/database/tasks/ongoingTaskOlapEtlTransformationModel.ts [hunk 1]
```diff
@@ -63,7 +65,7 @@ class ongoingTaskOlapEtlTransformationModel {
             ApplyToAllDocuments: this.applyScriptForAllCollections(),
             Collections: this.applyScriptForAllCollections() ? null : this.transformScriptCollections(),
             Disabled: false,
-            Name: this.name(),
+            Name: this.name().trim(),
```
Comment (ml054) [inline src/Raven.Studio/typescript/models/database/tasks/ongoingTaskOlapEtlTransformationModel.ts]: do we want that? usually we don't trim props.

Comment (ml054) [reply src/Raven.Studio/typescript/models/database/tasks/ongoingTaskOlapEtlTransformationModel.ts]: if someone save task outside studio with space then we effectively edit name by just saving in studio

File: src/Raven.Studio/typescript/models/database/tasks/ongoingTaskOlapEtlTransformationModel.ts [hunk 2]
```diff
@@ -74,6 +76,31 @@ class ongoingTaskOlapEtlTransformationModel {
...
+                new validateNameCommand('Script', val)
+                    .execute()
+                    .done((result) => {
+                        if (result.IsValid) {
+                            callback(this.name(), true);
+                        } else {
+                            callback(this.name(), result.ErrorMessage);
+                        }
+                    })
+            }
```
Comment (ml054) [inline src/Raven.Studio/typescript/models/database/tasks/ongoingTaskOlapEtlTransformationModel.ts]: do we want to call `callback` then name is null or empty?

File: src/Raven.Studio/typescript/models/database/tasks/ongoingTaskRavenEtlTransformationModel.ts [hunk 3]
```diff
@@ -58,7 +58,7 @@ class ongoingTaskRavenEtlTransformationModel {
             ApplyToAllDocuments: this.applyScriptForAllCollections(),
             Collections: this.applyScriptForAllCollections() ? null : this.transformScriptCollections(),
             Disabled: false,
-            Name: this.name(),
+            Name: this.name().trim(),
```
Comment (ml054) [inline src/Raven.Studio/typescript/models/database/tasks/ongoingTaskRavenEtlTransformationModel.ts]: same here

File: src/Raven.Studio/wwwroot/App/views/database/tasks/editOlapEtlTask.html [hunk 4]
```diff
@@ -221,81 +221,81 @@ <h3 class="margin-top margin-top-sm">The default ID column name is <code>_id</co
...
-                    <div class="padding padding-sm">
-                        <div data-bind="visible: !isNew()">
-                            <label>
-                                <strong>Name:</strong> <span data-bind="text: name"></span>
-                            </label>
+                <div class="panel padding" data-bind="if: $root.showEditTransformationArea">
+                    <div class="flex-horizontal margin-bottom" data-bind="validationElement: name">
+                        <label class="control-label"><strong>Name:</strong></label>
+                        <div class="flex-grow margin-left">
+                            <input type="text" class="form-control" placeholder="Enter script name (optional)" autocomplete="off"
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/App/views/database/tasks/editOlapEtlTask.html]: name isn't optional - we just autogenerate them.

File: src/Raven.Studio/typescript/viewmodels/database/tasks/editRavenEtlTask.ts [hunk 5]
```diff
@@ -439,7 +439,7 @@ class editRavenEtlTask extends viewModelBase {
         
         if (transformation.isNew()) {
             const newTransformationItem = new ongoingTaskRavenEtlTransformationModel(transformation.toDto(), true, false); 
-            newTransformationItem.name(this.findNameForNewTransformation());
+            newTransformationItem.name(transformation.name() || this.findNameForNewTransformation());
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/tasks/editRavenEtlTask.ts]: we should assign and display this name when user enters the view.

File: src/Raven.Studio/typescript/models/database/tasks/ongoingTaskOlapEtlTransformationModel.ts (hunk 1 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/models/database/tasks/ongoingTaskOlapEtlTransformationModel.ts]: removed trim

File: src/Raven.Studio/typescript/models/database/tasks/ongoingTaskRavenEtlTransformationModel.ts (hunk 3 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/models/database/tasks/ongoingTaskRavenEtlTransformationModel.ts]: removed

Comment (Danielle9897) [reply src/Raven.Studio/typescript/models/database/tasks/ongoingTaskRavenEtlTransformationModel.ts:61]: removed

File: src/Raven.Studio/typescript/models/database/tasks/ongoingTaskOlapEtlTransformationModel.ts (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/models/database/tasks/ongoingTaskOlapEtlTransformationModel.ts]: there is a check:  ` if (this.name())`  few lines above

File: src/Raven.Studio/wwwroot/App/views/database/tasks/editOlapEtlTask.html (hunk 4 above)
Comment (Danielle9897) [reply src/Raven.Studio/wwwroot/App/views/database/tasks/editOlapEtlTask.html]: ok

File: src/Raven.Studio/typescript/viewmodels/database/tasks/editRavenEtlTask.ts (hunk 5 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/tasks/editRavenEtlTask.ts]: done - showing "Script_1" immediately - upon opening view

File: src/Raven.Studio/typescript/models/database/tasks/ongoingTaskOlapEtlTransformationModel.ts [hunk 6]
```diff
@@ -47,13 +49,13 @@ class ongoingTaskOlapEtlTransformationModel {
        ], false, jsonUtil.newLineNormalizingHashFunction);
     }
    
-    static empty(): ongoingTaskOlapEtlTransformationModel {
+    static empty(suggestedName: string): ongoingTaskOlapEtlTransformationModel {
```
Comment (ml054) [inline src/Raven.Studio/typescript/models/database/tasks/ongoingTaskOlapEtlTransformationModel.ts]: it's simply name  - not suggested name. Suggested name is an example of usage.

File: src/Raven.Studio/typescript/models/database/tasks/ongoingTaskOlapEtlTransformationModel.ts (hunk 2 above)
Comment (ml054) [reply src/Raven.Studio/typescript/models/database/tasks/ongoingTaskOlapEtlTransformationModel.ts]: yes I know but the question is if we should call callback even when name is empty?

File: src/Raven.Studio/typescript/models/database/tasks/ongoingTaskRavenEtlTransformationModel.ts [hunk 7]
```diff
@@ -42,13 +42,13 @@ class ongoingTaskRavenEtlTransformationModel {
         return collectionsTracker.default.getCollectionColorIndex(collectionName);
     }
 
-    static empty(): ongoingTaskRavenEtlTransformationModel {
+    static empty(suggestedName: string): ongoingTaskRavenEtlTransformationModel {
```
Comment (ml054) [inline src/Raven.Studio/typescript/models/database/tasks/ongoingTaskRavenEtlTransformationModel.ts]: rename

File: src/Raven.Studio/typescript/models/database/tasks/ongoingTaskSqlEtlTransformationModel.ts [hunk 8]
```diff
@@ -33,13 +33,13 @@ class ongoingTaskSqlEtlTransformationModel {
         ], false, jsonUtil.newLineNormalizingHashFunction);
     }
    
-    static empty(): ongoingTaskSqlEtlTransformationModel {
+    static empty(suggestedName: string): ongoingTaskSqlEtlTransformationModel {
```
Comment (ml054) [inline src/Raven.Studio/typescript/models/database/tasks/ongoingTaskSqlEtlTransformationModel.ts]: rename

File: src/Raven.Studio/typescript/models/database/tasks/ongoingTaskOlapEtlTransformationModel.ts (hunk 6 above)
Comment (ml054) [reply src/Raven.Studio/typescript/models/database/tasks/ongoingTaskOlapEtlTransformationModel.ts]: additionally we might made this parameter optional

File: src/Raven.Studio/typescript/viewmodels/database/tasks/editOlapEtlTask.ts [hunk 9]
```diff
@@ -656,8 +656,10 @@ class editOlapEtlTask extends viewModelBase {
 
         if (transformation.isNew()) {
             const newTransformationItem = new ongoingTaskOlapEtlTransformationModel(transformation.toDto(), false, false);
-            newTransformationItem.name(this.findNameForNewTransformation());
+            
+            newTransformationItem.name(transformation.name() || this.findNameForNewTransformation());
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/tasks/editOlapEtlTask.ts]: do we need that? Shouldn't we add validation that name is required?

File: src/Raven.Studio/typescript/viewmodels/database/tasks/editRavenEtlTask.ts (hunk 5 above)
Comment (ml054) [reply src/Raven.Studio/typescript/viewmodels/database/tasks/editRavenEtlTask.ts]: do we need that? At this point name should be valid (not empty)

File: src/Raven.Studio/typescript/viewmodels/database/tasks/editSqlEtlTask.ts [hunk 10]
```diff
@@ -624,7 +624,7 @@ class editSqlEtlTask extends viewModelBase {
 
         if (transformation.isNew()) {
             const newTransformationItem = new ongoingTaskSqlEtlTransformationModel(transformation.toDto(), false, false);
-            newTransformationItem.name(this.findNameForNewTransformation());
+            newTransformationItem.name(transformation.name() || this.findNameForNewTransformation());
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/tasks/editSqlEtlTask.ts]: same here

File: src/Raven.Studio/typescript/models/database/tasks/ongoingTaskOlapEtlTransformationModel.ts (hunk 6 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/models/database/tasks/ongoingTaskOlapEtlTransformationModel.ts]: done

File: src/Raven.Studio/typescript/models/database/tasks/ongoingTaskRavenEtlTransformationModel.ts (hunk 7 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/models/database/tasks/ongoingTaskRavenEtlTransformationModel.ts]: done

File: src/Raven.Studio/typescript/models/database/tasks/ongoingTaskSqlEtlTransformationModel.ts (hunk 8 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/models/database/tasks/ongoingTaskSqlEtlTransformationModel.ts]: done

File: src/Raven.Studio/typescript/viewmodels/database/tasks/editOlapEtlTask.ts (hunk 9 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/tasks/editOlapEtlTask.ts]: Don't we want to give users **the option** to leave the script name blank and assign 'Script_x' for the user ?

Comment (ml054) [reply src/Raven.Studio/typescript/viewmodels/database/tasks/editOlapEtlTask.ts]: no - we fill that initially and then require name

Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/tasks/editOlapEtlTask.ts]: ok done

File: src/Raven.Studio/typescript/viewmodels/database/tasks/editRavenEtlTask.ts (hunk 5 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/tasks/editRavenEtlTask.ts]: ok - removed

File: src/Raven.Studio/typescript/viewmodels/database/tasks/editSqlEtlTask.ts (hunk 10 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/tasks/editSqlEtlTask.ts]: done

File: src/Raven.Studio/typescript/viewmodels/database/tasks/editOlapEtlTask.ts (hunk 9 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/tasks/editOlapEtlTask.ts:660]: done

File: src/Raven.Studio/typescript/models/database/tasks/ongoingTaskOlapEtlTransformationModel.ts (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/models/database/tasks/ongoingTaskOlapEtlTransformationModel.ts]: added `required: true` so now this callback is not being called if empty

**Review by ml054 — APPROVED** [review]

**Pull request merged**